kong/router/traditional.lua

```diff
@@ -1530,11 +1530,11 @@ function _M.new(routes, cache, cache_neg)
                                  .. "|" .. sni .. headers_key
     local match_t = cache:get(cache_key)
     if match_t then
-      return match_t
+      return match_t, "pos"
```

Comment:
instead of using a second return parameter (usually the place for `err`), consider return it in match_t to avoid confusion. for neg match, we might return a global static table.

on the other side, what's the reason of not setting `ctx.route_match_cached = xxx` inside find_route?